go-controller/pkg/ovn/ovn_test.go

```diff
@@ -54,9 +54,18 @@ type FakeOVN struct {
 	egressSVCWg  *sync.WaitGroup
 }
 
-func NewFakeOVN() *FakeOVN {
+// NOTE: the FakeAddressSetFactory is no longer needed and should no longer be used. starting to phase out FakeAddressSetFactory
+func NewFakeOVN(useFakeAddressSet bool) *FakeOVN {
+	if useFakeAddressSet {
```

Comment:
@JacobTanenbaum instead of copying the whole FakeOVN thing, I'd just do:

var asf *addressset.FakeAddressSetFactory
if useFakeAddressSet {
    asf = addressset.NewFakeAddressSetFactory(DefaultNetworkControllerName)
}

...

return &FakeOVN{
		asf:          asf,